simplesat/compute_dependencies.py

```diff
@@ -63,6 +63,42 @@ def compute_reverse_dependencies(repositories, requirement, transitive=False):
     return dependencies
 
 
+def compute_leaf_packages(repositories):
+    """ Compute the leaf packages in `repositories`. Leaf packages are packages
+    with no reverse dependencies.
+
+    Parameters
+    ----------------
+    repositories : iterable of Repository objects
+
+    Returns
+    -----------
+    dependencies : set
+        Set of leaf packages in the given repositories.
+    """
+    pool = Pool(repositories)
+    package_ids = set(pool.iter_package_ids())
+    neighbors = _compute_dependency_dict(pool, package_ids, transitive=False)
+
+    # Reverse mapping so that package ids point to the packages which depend
+    # on them
+    reverse_neighbors = _reverse_mapping(neighbors)
+
+    leaf_packages = set()
+    for repository in repositories:
+        for package in repository:
+            package_string = package.name + "-" + str(package.version)
```

Comment:
It seems like you ought to be able to create the `InstallRequirement` directly from the `PackageMetadata`, but I can't find a way to do so.